src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,137 @@
+        # To initialize the queue in A*, we will build up a partial policy
+        # and then run policy-guided A*.
+        start_time = time.time()
+        objects = set(state)
+        partial_policy_dict: Dict[FrozenSet[GroundAtom], _GroundNSRT] = {}
+        for option_plan in self._get_llm_based_option_plans(
+                atoms, objects, goal):
+            ground_nsrt_plan = self._option_plan_to_nsrt_plan(
+                option_plan, atoms, objects, goal)
+            assert ground_nsrt_plan is not None
+            # Simulate the ground NSRT plan and record the (atoms, ground NSRT)
+            # pairs encountered along the way to build up a partial policy.
+            single_partial_policy_dict = {}
+            cur_atoms = atoms
+            for ground_nsrt in ground_nsrt_plan:
+                # If we run into an invalid ground NSRT, the rest of the plan
+                # is going to be garbage, so stop here.
+                if not ground_nsrt.preconditions.issubset(cur_atoms):
+                    break
+                single_partial_policy_dict[frozenset(cur_atoms)] = ground_nsrt
+                cur_atoms = utils.apply_operator(ground_nsrt, cur_atoms)
+            # Update the overall partial policy. Note that there may be
+            # entries that get overridden. This is okay, and there's no
+            # reason to prefer one option plan over another, so we just let
+            # the overriding happen.
+            partial_policy_dict.update(single_partial_policy_dict)
+        # Run policy-guided A*. Note that we importantly want to use
+        # sesame_plan because we will want to report the metrics collected in
+        # sesame_plan.
+        abstract_policy = lambda a, o, g: partial_policy_dict.get(
+            frozenset(a), None)
+        max_policy_guided_rollout = CFG.horizon  # not important
+        nsrts = self._get_current_nsrts()
+        preds = self._get_current_predicates()
+        task = Task(state, goal)
+        options, metrics = self._run_sesame_plan(
+            task,
+            nsrts,
+            preds,
+            CFG.timeout - (time.time() - start_time),
+            CFG.seed,
+            abstract_policy=abstract_policy,
+            max_policy_guided_rollout=max_policy_guided_rollout)
+        self._save_metrics(metrics, nsrts, preds)
+        # Now convert the options back into ground NSRTs (:facepalm:). This
+        # is very circuitous and we should refactor it later. There are two
+        # original sins that are converging and making this hard: (1) the
+        # fact that the skeleton generator implements its own search, because
+        # otherwise we could just run the search utility functions here; and
+        # (2) the fact that this and the parent class are implemented as
+        # metacontrollers, which do not use sesame plan.
+        option_plan = [(o.parent, o.objects) for o in options]
```

Comment:
so this option_plan is the one that comes out of sesame, and therefore is guaranteed to be a valid solution (as long as our option model is accurate) -- right? (i'm contrasting this with the option_plans that you looped over above, `for option_plan in self._get_llm_based_option_plans()`, which have no guarantees)

Reply:
right